sentry-core/src/main/java/io/sentry/core/SentryEnvelopeHeaderAdapter.java

```diff
@@ -35,20 +71,51 @@ public SentryEnvelopeHeader read(JsonReader reader) throws IOException {
     }
 
     SentryId eventId = null;
+    SdkInfo sdkInfo = null;
 
     reader.beginObject();
     while (reader.hasNext()) {
       switch (reader.nextName()) {
         case "event_id":
           eventId = new SentryId(reader.nextString());
           break;
+        case "sdk_info":
+          {
+            reader.beginObject();
+            sdkInfo = new SdkInfo();
+
+            while (reader.hasNext()) {
+              switch (reader.nextName()) {
+                case "sdk_name":
+                  sdkInfo.setSdkName(reader.nextString());
+                  break;
+                case "version_major":
+                  sdkInfo.setVersionMajor(reader.nextInt());
+                  break;
+                case "version_minor":
+                  sdkInfo.setVersionMinor(reader.nextInt());
+                  break;
+                case "version_patchlevel":
+                  sdkInfo.setVersionPatchlevel(reader.nextInt());
+                  break;
+                default:
+                  reader.skipValue();
```

Comment:
M: Ideally we'd keep this in a bag (a Map<string, object> or something) to serialize back so it's future proof.
If sentry-native decides to add more data, we'd be dropping those when deserializing

Reply:
this already works for the SdkInfo object in an event, this is specifically for the envelope header, do we need more than the name and version? I mean it's fine to do it but it looks like not useful, wdyt?